File: Maria Mercedes Atim/JAVA/Clase 9/Clase 9/src/domain/Empleado.java
package domain;

public class Empleado extends Persona {

    private int idEmpleado; // Identificador único de cada empleado
    private double sueldo; // Sueldo del empleado
    private static int contadorEmpleados; // Contador estático para generar idEmpleado únicos

    // Constructores de la clase Empleado

    public Empleado() {
        this.idEmpleado = ++Empleado.contadorEmpleados; // Incrementa y asigna el idEmpleado
    }
    
    
    public Empleado(String nombre, double sueldo) {
        // super(nombre); // Llama al constructor de la clase padre (Persona)
        this(); //Estamos llamando al constructor interno
        this.nombre = nombre;
        this.sueldo = sueldo; // Asigna el sueldo del empleado
    }

    // Getter para obtener el sueldo
    public double getSueldo() {
        return sueldo;
    }

    // Getter para obtener el id del empleado
    public int getIdEmpleado() {
        return idEmpleado;
    }

    // Setter para modificar el sueldo
    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }   

    // Sobrescribe el método toString para imprimir detalles del empleado
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder(); // Usa StringBuilder para crear la cadena
        sb.append("Empleado ");
        sb.append(" idEmpleado: ").append(idEmpleado); // Agrega el idEmpleado
        sb.append("\nsueldo: ").append(sueldo); // Agrega el sueldo
        sb.append("\n").append(super.toString()); // Incluye los detalles de la clase Persona
        return sb.toString(); // Retorna la cadena con los detalles del empleado
    }
}

File: TECNICATURA/JAVASCRIPT/Clase09/Lección07/Empleado.js
class Empleado extends Persona {
    static contadorEmpleados = 0;

    constructor(nombre, apellido, edad, sueldo) {
        super(nombre, apellido, edad);
        this._idEmpleado = ++Empleado.contadorEmpleados;
        this._sueldo = sueldo;
    }

    // Getter para idEmpleado
    get idEmpleado() {
        return this._idEmpleado;
    }

    // Getter para sueldo
    get sueldo() {
        return this._sueldo;
    }

    // Setter para sueldo
    set sueldo(nuevoSueldo) {
        if (nuevoSueldo >= 0) {
            this._sueldo = nuevoSueldo;
        } else {
            console.log("El sueldo no puede ser negativo.");
        }
    }

    // Método toString
    toString() {
        return `${super.toString()}, Empleado ID: ${this._idEmpleado}, Sueldo: ${this._sueldo}`;
    }
}